### PR TITLE
docs(utils): replace any type to specified type in minBy doc

### PR DESCRIPTION
**Pull request by euijinkk**
## Overview

- replace any type to specified type in minBy function JSDoc

I found other mismatch between JSDoc and code. 
slash team think it should be edited? And I wonder if slash team made JSDOc one by one or not.

## PR Checklist

- [x] I read and included theses actions below

1. I have read the [Contributing Guide](https://github.com/toss/slash/blob/main/.github/CONTRIBUTING.md)
2. I have written documents and tests, if needed.

### CODE REVIEW

File: packages/common/utils/src/minBy.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@
  *   // 최솟값을 찾을 배열
  *   collection: T[],
  *   // 배열의 값을 계산하는 방법
- *   iteratee: (element: T) => any
+ *   iteratee: (element: T) => T | undefined
```
Comment (ChanhyukPark-Tech) [inline packages/common/utils/src/minBy.ts:10]: I think not this should be fixed to 
```suggestion
 *   iteratee: (element: T) => number
```

according to code itself return type of iteratee function is number not T | undefined

Comment (euijinkk) [reply packages/common/utils/src/minBy.ts]: Sorry, that's my mistake. 
I fixed it according to your suggestion.

[docs: fix iteratee return type](https://github.com/toss/slash/pull/59/commits/dd5395d55ee60624e61e8c9f3bed46ffbc107742)

**Comment by raon0211** [comment]
Actually we manually have written all the jsdocs so there may be some mistakes 😢  thanks for your contribution!

**Review by raon0211 — APPROVED** [review]
Thanks a lot, @euijinkk ! 🙌

**Pull request merged**